src/components/TabletApp/RightSide/QuickBook/QuickBook.tsx

```diff
@@ -4,53 +4,94 @@ import {
-
-//To be removed when globla theme is done
-const Item = styled(Paper)(({ theme }) => ({
-    backgroundColor: theme.palette.mode === "dark" ? "#1A2027" : "#fff",
-    ...theme.typography.body2,
-    padding: theme.spacing(1),
-    textAlign: "center",
-    color: theme.palette.text.secondary,
-}));
-
-//To be removed when global theme is done
-const buttonStyle = { color: COLORS.GREEN, border: `1px solid ${COLORS.GREEN}` };
+import { getParticipants, getParticipant } from "../../../../api/getRequests";
+import dayjs from "dayjs";
+import { spawnToast } from "../../../../utils/Toast";
+
+const Owner = {
+    name: "",
+    id: 0,
+    available: true,
+    image: "",
+};
+interface MeetTime {
+    end_time: string;
+    start_time: string;
+}
 
 const QuickBook = () => {
     const [timeButtonsVisible, setTimeButtonsVisible] = useState(false);
     const [open, setOpen] = useState(false);
-    const [ownerName, setOwnerName] = useState("");
+    const [owner, setOwner] = useState(Owner);
     const [owners, setOwners] = useState([""]);
     const [autoComplete, setAutoComplete] = useState(false);
+    const [time_val, setTime_val] = useState(0);
+    const [closest_meet, setMeet] = useState(0);
+    const [in_session, set_in_session] = useState(false);
 
-    const handleQuickBook = () => {
+    const handleQuickBook = async () => {
+        set_in_session(false);
         setTimeButtonsVisible(!timeButtonsVisible);
         if (timeButtonsVisible) setOpen(false);
+
+        let meetings_start_time: any[] = [];
+        const response = await axios.get("http://localhost:3001/meetings");
+
+        Object.values(response.data as MeetTime[]).map((value) => {
+            if (dayjs(value.start_time).isSame(dayjs(), "day")) {
+                meetings_start_time.push(
+                    dayjs(value.start_time).diff(dayjs(), "minute")
+                );
+                if (dayjs(value.end_time).isAfter(dayjs())) {
+                    set_in_session(true);
+                }
+            }
+        });
+
+        meetings_start_time.sort((a, b) =>
+            dayjs(a).isAfter(dayjs(b)) ? -1 : 1
+        );
+
+        setMeet(meetings_start_time[0]);
     };
 
     const handleClickTime = () => {
         if (!open) setOpen(true);
-        setOwnerName("");
+        setOwner(Owner);
     };
 
-    const handleChange = (e: any) => {
-        setOwnerName(e.target.value);
+    const handleChange = async (e: any) => {
+        const result = await getParticipant(e.target.innerHTML);
+        setOwner(result.data[0]);
     };
 
-    const handleSearchOwner = () => {
-        console.log(ownerName);
-        //To be implemented. Final step, needs to create meeting
+    const handleSubmitOwner = async () => {
+        let now = dayjs();
+        let end_time = now.add(time_val, "minute");
+        let latest_meetings_id = 0;
+
+        const response = await axios.get("http://localhost:3001/meetings");
```

Comment:
I see you are doing GET "/meetings" multiple times in this component and you should only do it once the component renders and save the response into a state and use that state throughout the component

Reply:
I need the repeated get in the handleQuickBookButton to check if there's other meetings that are still running and haven't finished yet, so that the quick button is disabled in that case.

Reply:
Added all the get calls to a useEffect